CRM/Financeextras/BAO/CreditNoteLine.php

```diff
@@ -22,4 +24,89 @@ public static function create($params) {
+   *  The credit note entity to create line items for
+   *
+   * @param array $financialTrxn
+   *  The credit note financial transaction entity.
+   *
+   * @return array
+   *   Created line items.
+   */
+  public static function createWithAcountingEntries($items, $creditNote, $financialTrxn) {
+    $result = [];
+    array_walk($items, function (&$lineItem) use ($creditNote, $financialTrxn, &$result) {
+      $lineTotal = $lineItem['unit_price'] * $lineItem['quantity'];
+      $lineItemParams = [
+        'credit_note_id' => $creditNote['id'],
+        'description' => $lineItem['description'],
+        'quantity' => $lineItem['quantity'],
+        'unit_price' => $lineItem['unit_price'],
+        'tax_amount' => empty($lineItem['tax_rate']) ? 0 : self::percent($lineItem['tax_rate'], $lineTotal),
+        'line_total' => $lineTotal,
+        'financial_type_id' => $lineItem['financial_type_id'],
+      ];
+
+      $lineItem = self::create($lineItemParams)->toArray();
+      self::createAccountingEntries($lineItem, $creditNote, $financialTrxn);
+
+      $result[] = $lineItem;
+    });
+
+    return $result;
+  }
+
+  private static function createAccountingEntries($lineItem, $creditNote, $financialTrxn) {
+    $incomeAccount = CreditNoteBAO::getFinancialTypeAccount($lineItem['financial_type_id'], 'Income Account is');
+    $itemParams = [
+      'transaction_date' => $creditNote['date'],
+      'contact_id' => $creditNote['contact_id'],
+      'currency' => $creditNote['currency'],
+      'amount' => ($lineItem['quantity'] * $lineItem['unit_price']) * -1,
+      'description' => $lineItem['description'],
+      'status_id' => 'Unpaid',
+      'financial_account_id' => $incomeAccount,
+      'entity_table' => \CRM_Financeextras_DAO_CreditNoteLine::$_tableName,
+      'entity_id' => $lineItem['id'],
+    ];
+    \CRM_Financial_BAO_FinancialItem::create($itemParams, NULL, $financialTrxn);
+
+    if (!empty($lineItem['tax_amount']) && $lineItem['tax_amount'] > 0) {
+      $taxAccount = CreditNoteBAO::getFinancialTypeAccount($lineItem['financial_type_id'], 'Sales Tax Account is');
+      $taxAccountDesc = \Civi\Api4\FinancialAccount::get()
+        ->addSelect('description')
+        ->addWhere('id', '=', $taxAccount)
+        ->execute()
+        ->first()['description'] ?? "";
+
+      $itemParams['amount'] = $lineItem['tax_amount'] * -1;
+      $itemParams['financial_account_id'] = $taxAccount;
+      $itemParams['description'] = $lineItem['description'] . " - " . $taxAccountDesc;
+      \CRM_Financial_BAO_FinancialItem::create($itemParams, NULL, $financialTrxn);
+    }
+  }
+
+  /**
+   * Returns percentage of value.
+   *
+   * E.g. 5% of 10.
+   *
+   * @param float $percentage
+   *   Percentage to calculate.
+   * @param float $value
+   *   The value to get percentage of.
+   *
+   * @return float
+   *   Calculated Percentage in float
+   */
+  private static function percent(float $percentage, float $value) {
```

Comment:
a better name for this might be calculateTaxAmount() with this signature:

```
private static function calculateTaxAmount(float $amount, float $taxRate)
```